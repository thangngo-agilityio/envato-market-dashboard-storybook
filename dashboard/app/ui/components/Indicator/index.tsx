import { Center, Spinner } from '@chakra-ui/react';
import { ReactNode } from 'react';

type TIndicatorProps = {
  isOpen?: boolean;
  children?: ReactNode;
};

const Indicator = ({
  isOpen = false,
  children,
}: TIndicatorProps): JSX.Element => (
  <>
    {children}
    {isOpen && (
      <Center
        position="fixed"
        zIndex={999999999}
        inset={0}
        bg="black"
        opacity={0.7}
      >
        <Spinner
          size={{
            base: 'md',
            md: 'lg',
            lg: 'xl',
          }}
          color="blue.500"
        />
      </Center>
    )}
  </>
);

export default Indicator;
