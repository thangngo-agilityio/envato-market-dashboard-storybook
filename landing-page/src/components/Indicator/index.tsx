import type { ReactNode } from 'react';

type TIndicatorProps = {
  isOpen?: boolean;
  children?: ReactNode;
};

const Indicator = ({
  isOpen = false,
  children,
}: TIndicatorProps): JSX.Element => (
  <>
    {children}
    {isOpen && (
      <section className='fixed flex justify-center items-center z-50 inset-0 bg-black opacity-65'>
        <span className='block w-10 h-10 bg-transparent border-2 border-gray-50 rounded-full border-l-transparent animate-spinner'></span>
      </section>
    )}
  </>
);

export default Indicator;
