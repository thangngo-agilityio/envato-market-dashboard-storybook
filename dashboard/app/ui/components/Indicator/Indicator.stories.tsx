import { StoryObj, Meta } from '@storybook/react';

// Components
import { Indicator } from '@/ui/components';
import { Box } from '@chakra-ui/react';

const meta: Meta<typeof Indicator> = {
  title: 'Custom Components/Indicator',
  component: Indicator,
  tags: ['autodocs'],
};

export default meta;
type Story = StoryObj<typeof Indicator>;

export const Default: Story = {
  args: {
    isOpen: true,
    children: <Box py={16} />,
  },
};
