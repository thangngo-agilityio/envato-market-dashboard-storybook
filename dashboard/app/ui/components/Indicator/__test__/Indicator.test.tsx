import { render } from '@testing-library/react';

// Component
import Indicator from '..';

describe('Indicator components', () => {
  it('match snapshot with open true', () => {
    const { container } = render(
      <Indicator isOpen={true}>Test Indicator</Indicator>,
    );

    expect(container).toMatchSnapshot();
  });
});
