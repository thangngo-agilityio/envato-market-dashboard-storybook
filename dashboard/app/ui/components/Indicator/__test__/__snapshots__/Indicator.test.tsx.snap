// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Indicator components match snapshot with open true 1`] = `
<div>
  Test Indicator
  <div
    class="css-q6zik0"
  >
    <div
      class="chakra-spinner css-75d0f3"
    >
      <span
        class="css-8b45rq"
      >
        Loading...
      </span>
    </div>
  </div>
</div>
`;
